src/openforms/js/components/admin/form_design/registrations/zgw/fields/MedewerkerRoltype.js

```diff
@@ -1,12 +1,87 @@
-import {useField} from 'formik';
+import {useField, useFormikContext} from 'formik';
+import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
+import useAsync from 'react-use/esm/useAsync';
 
 import Field from 'components/admin/forms/Field';
 import FormRow from 'components/admin/forms/FormRow';
 import {TextInput} from 'components/admin/forms/Inputs';
+import ReactSelect from 'components/admin/forms/ReactSelect';
+import {get} from 'utils/fetch';
 
-const MedewerkerRoltype = () => {
+// Data fetching
+
+const ROLE_TYPES_ENDPOINT = '/api/v2/registration/plugins/zgw-api/role-types';
+
+const getAvailableRoleTypes = async (apiGroupID, catalogueUrl, caseTypeIdentification) => {
+  const response = await get(ROLE_TYPES_ENDPOINT, {
+    zgw_api_group: apiGroupID,
+    catalogue_url: catalogueUrl,
+    case_type_identification: caseTypeIdentification,
+  });
+  if (!response.ok) {
+    throw new Error('Loading available object types failed');
+  }
+  const roleTypes = response.data.sort((a, b) => a.description.localeCompare(b.description));
+  return roleTypes.map(({description, descriptionGeneric}) => ({
+    value: description,
+    label: description,
+    descriptionGeneric: descriptionGeneric, // omschrijvingGeneriek, which is an enum
```

Comment:
Is `descriptionGeneric` this needed?

Reply:
not used at the moment, though I do anticipate that it'll possibly be displayed similarly to a variable key. I've opted to include it here so that it's obvious that we *do have* that information.